pkg/cmd/container/unpause.go

```diff
@@ -0,0 +1,54 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package container
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/nerdctl/pkg/api/types"
+	"github.com/containerd/nerdctl/pkg/containerutil"
+	"github.com/containerd/nerdctl/pkg/idutil/containerwalker"
+)
+
+// Unpause unpauses all containers specified by `reqs`.
+func Unpause(ctx context.Context, client *containerd.Client, reqs []string, options types.ContainerUnpauseOptions) error {
+	walker := &containerwalker.ContainerWalker{
+		Client: client,
+		OnFound: func(ctx context.Context, found containerwalker.Found) error {
+			if found.MatchCount > 1 {
+				return fmt.Errorf("multiple IDs found with provided prefix: %s", found.Req)
+			}
+			if err := containerutil.Unpause(ctx, client, found.Container.ID()); err != nil {
+				return err
+			}
+
+			_, err := fmt.Fprintf(options.Stdout, "%s\n", found.Req)
+			return err
+		},
+	}
+	for _, req := range reqs {
+		n, err := walker.Walk(ctx, req)
+		if err != nil {
+			return err
+		} else if n == 0 {
+			return fmt.Errorf("no such container %s", req)
+		}
```

Comment:
would this fail fast and is that expected? i.e., an alternative would be notifying the error but continuing to walk reqs and attempt to unpause those, perhaps returning multierror

Reply:
yeah I also prefer to proceed to the next and print all errors at the end. 

I'm planning to keep the implementation unchanged in this refactor PR and fix in a follow-up PR, since it's not only in (un)pause, we have similar logic to update in `kill`/`start`/`kill`/`update` and some image subcommands.

Reply:
And docker also did this printing errors at the end: https://github.com/docker/cli/blob/e92dd87c3209361f29b692ab4b8f0f9248779297/cli/command/container/kill.go#L50-L59

Reply:
:) decided to update (un)pause to continue to walk and return all errors (combined) at the end if there is any. Didn't update compose (un)pause since it requires more changes and the current behavior matches docker compose:

https://github.com/docker/compose/blob/1640f155e94468e76ccb0b4f16c9592983f22aec/pkg/compose/pause.go#L47-L59